test/NuGet.Core.Tests/NuGet.Packaging.Test/PackageBuilderTest.cs

```diff
@@ -371,6 +374,42 @@ public void CreatePackageWithNuspecIncludeExcludeWithWildcards(string source, st
             }
         }
 
+        [PlatformTheory(Platform.Windows)]
+        [InlineData(@"dir1\dir2\**", true, "Content")]
+        [InlineData("dir1/dir2/**", false, "Content")]
+        public void PackageBuilder_AddFiles_HasDifferentBehaviorDependingOnSlash(string source, bool expectFlattened, string destination)
+        {
+            // https://github.com/NuGet/Home/issues/11234
+            using (var directory = new TestSourcesDirectory())
+            {
+                // Arrange
+                PackageBuilder builder = new PackageBuilder();
+
+                // Act
+                builder.AddFiles(directory.Path, source, destination);
+
+                // Assert
+                var expectedResults = new[]
+                {
+                    new
+                    {
+                        Path = expectFlattened ? $"Content{Path.DirectorySeparatorChar}file1.txt" : $"Content{Path.DirectorySeparatorChar}dir1{Path.DirectorySeparatorChar}dir2{Path.DirectorySeparatorChar}file1.txt",
+                        EffectivePath = expectFlattened ? "file1.txt" : $"dir1{Path.DirectorySeparatorChar}dir2{Path.DirectorySeparatorChar}file1.txt"
+                    },
+                    new
+                    {
+                        Path = expectFlattened ? $"Content{Path.DirectorySeparatorChar}file2.txt" : $"Content{Path.DirectorySeparatorChar}dir1{Path.DirectorySeparatorChar}dir2{Path.DirectorySeparatorChar}file2.txt",
+                        EffectivePath = expectFlattened ? "file2.txt" : $"dir1{Path.DirectorySeparatorChar}dir2{Path.DirectorySeparatorChar}file2.txt"
+                    }
+                };
+
+                var orderedExpectedResults = expectedResults.OrderBy(i => i.Path);
+                var orderedActualResults = builder.Files.Select(f => new { f.Path, f.EffectivePath }).OrderBy(i => i.Path);
```

Comment:
explicit types per rule 10: https://github.com/NuGet/NuGet.Client/blob/dev/docs/coding-guidelines.md#c-coding-style

Reply:
I copied this from above, the result is an enumerable of dynamic types declared above.  In this particular case there is no way to declare the type.

https://github.com/NuGet/NuGet.Client/blob/1af032a9a12c530ef434e9ea1a53cd93348e1146/test/NuGet.Core.Tests/NuGet.Packaging.Test/PackageBuilderTest.cs#L367